### ISSUE TITLE
Unexpected "Expression is always true" error.

### ISSUE THREAD

**Issue by vilic** [palantir/tslint#2472]
### Bug Report

- __TSLint version__: 5.0.0
- __TypeScript version__: 2.2.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
class Foo { }

function foo(p: typeof Foo | object) {
  if (typeof p === 'object') {
    // ...
  }
}
```

With `strict-type-predicates` enabled.

#### Actual behavior

ERROR: Expression is always true.

#### Expected behavior

No error.

**Comment by andy-hanson** [comment]
This depends on Microsoft/TypeScript#13502. `typeof Foo` is assignable to `Function`, but it's not itself a function type.

**Comment by vilic** [comment]
@andy-hanson I am not familiar with that, what do you mean by "function type"? Does `typeof Foo` have a signature of `new (): Foo;`?

**Comment by andy-hanson** [comment]
Actually, this is doable if we look for the construct signature.
